src/main/java/apoc/coll/Coll.java

```diff
@@ -828,4 +828,33 @@ public boolean different(@Name("values") List<Object> values) {
 		if (values == null) return false;
         return new HashSet(values).size() == values.size();
     }
+
+
+
+    @Procedure
+    @Description("apoc.coll.dropDuplicateNeighbors(list) - remove duplicate consecutive objects in a list")
+    public Stream<ListResult> dropDuplicateNeighbors(@Name("list") List<Object> list) {
+        if (sameTypeForAllElements(list)) {
```

Comment:
Con questo if cicliamo due volte la collezione, la prima per verificare che gli elementi siano dello stesso tipo e la seconda per filtrare i duplicati.
Possiamo fare tutto in un unico cliclo? Esempio:
```
boolean hasSameType = true;
List<Object> newList = new ArrayList<>();
newList.add(list.get(0));
Class first = input.get(0).getClass();
for (int i = 1; i < list.size(); i++) {
        Object element = list.get(i);
        if (!element.equals(newList.get(newList.size() - 1))) {
                if (first.isAssignableForm(element.getClass()) {
                        newList.add(element);
                } else {
                        hasSameType = false;
                        break;
                } 
        }
}
if (!hasSameType) // throw dell'eccezione
```

Reply:
Così risparmiamo un cliclo e miglioriamo le performance nel caso di liste di grosse dimensioni